### PR TITLE
fix: restore support for grpcio-gcp

### PR DESCRIPTION
**Pull request by parthea**
This PR restores the feature which was removed in https://github.com/googleapis/python-api-core/pull/401 which is still used in spanner v1 [here](https://github.com/googleapis/python-spanner/blob/v1/setup.py#L32). `grpcio-gcp` is not compatible with protobuf 4.x.x so the feature is only supported with protobuf 4.x.x. This PR also adds a deprecation warning to let users know that support for `grpcio-gcp` will be dropped after January 1, 2024.

docs: add a note that grpcio-gcp is only supported in environments with protobuf < 4.x.x
docs: raise DeprecationWarning when 'grpcio-gcp' is used
fix(deps): require protobuf >= 3.20.1

### CODE REVIEW

File: google/api_core/grpc_helpers.py [hunk 1]
```diff
@@ -24,6 +25,27 @@
 import google.auth.credentials
 import google.auth.transport.grpc
 import google.auth.transport.requests
+import google.protobuf
+
+PROTOBUF_VERSION = google.protobuf.__version__
+
+# The grpcio-gcp package only has support for protobuf < 4
+if int(PROTOBUF_VERSION.split(".")[0]) < 4:
```
Comment (vam-google) [inline google/api_core/grpc_helpers.py:33]: Minor: guess we can just check if the first two characters are "4." no need to split and convert.

File: testing/constraints-3.7.txt [hunk 2]
```diff
@@ -6,9 +6,10 @@
 # e.g., if setup.py has "foo >= 1.14.0, < 2.0.0dev",
 # Then this file should have foo==1.14.0
 googleapis-common-protos==1.56.2
-protobuf==3.15.0
+protobuf==3.19.0
```
Comment (vam-google) [inline testing/constraints-3.7.txt:9]: Why not 3.21?

**Review by vam-google — APPROVED** [review]
LGTM, just a few minor things

File: google/api_core/grpc_helpers.py (hunk 1 above)
Comment (parthea) [reply google/api_core/grpc_helpers.py]: addressed in [6225d4f](https://github.com/googleapis/python-api-core/pull/418/commits/6225d4fad49792d88b00584f8c7596c2d02c076c)

File: testing/constraints-3.7.txt (hunk 2 above)
Comment (parthea) [reply testing/constraints-3.7.txt]: I've bumped the version to 3.20.1, the latest 3.x.x version
https://pypi.org/project/protobuf/3.20.1/

**Pull request merged**